src/viam/sdk/tests/test_resource.cpp

```diff
@@ -0,0 +1,172 @@
+#define BOOST_TEST_MODULE test module test_resource
+#include <boost/test/included/unit_test.hpp>
+
+#include <viam/sdk/common/pose.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/referenceframe/frame.hpp>
+#include <viam/sdk/resource/resource_api.hpp>
+#include <viam/sdk/spatialmath/geometry.hpp>
+#include <viam/sdk/spatialmath/orientation.hpp>
+#include <viam/sdk/spatialmath/orientation_types.hpp>
+
+namespace viam {
+namespace sdktests {
+
+using namespace viam::sdk;
+
+BOOST_AUTO_TEST_CASE(test_api) {
+    API api1;
+    api1.set_namespace("ns");
+    BOOST_CHECK_EQUAL(api1.type_namespace(), "ns");
+    api1.set_resource_type("component");
+    BOOST_CHECK_EQUAL(api1.resource_type(), "component");
+    api1.set_resource_subtype("st");
+    BOOST_CHECK_EQUAL(api1.resource_subtype(), "st");
+    BOOST_CHECK_EQUAL(api1.to_string(), "ns:component:st");
+    BOOST_CHECK(!api1.is_service_type());
+    BOOST_CHECK(api1.is_component_type());
+
+    API api2 = API::from_string("ns:service:st");
+    BOOST_CHECK_EQUAL(api2.to_string(), "ns:service:st");
+    BOOST_CHECK(api2.is_service_type());
+    BOOST_CHECK(!api2.is_component_type());
+
+    BOOST_CHECK_THROW(API::from_string("nsservicest"), std::string);
+    BOOST_CHECK_THROW(API::from_string("ns:service:#st"), std::string);
+    BOOST_CHECK_THROW(API::from_string("ns:service"), std::string);
+}
+
+BOOST_AUTO_TEST_CASE(test_name) {
+    Name name1(API::from_string("ns:service:st"), "remote", "name");
+    BOOST_CHECK_EQUAL(name1.api().to_string(), "ns:service:st");
+    BOOST_CHECK_EQUAL(name1.remote_name(), "remote");
+    BOOST_CHECK_EQUAL(name1.name(), "name");
+    BOOST_CHECK_EQUAL(name1.to_string(), "ns:service:st/remote:name");
+    BOOST_CHECK_EQUAL(name1.short_name(), "remote:name");
+    // TODO: Currently, remotes not being returned from proto for whatever reason
+    // BOOST_CHECK_EQUAL(Name::from_proto(name1.to_proto()), name1);
```

Comment:
Bug where remotes are not being stored or retrieved properly when `to_proto()` and `from_proto()` are called